x-pack/plugins/ml/public/embeddables/anomaly_swimlane/swimlane_input_resolver.ts

```diff
@@ -67,6 +67,23 @@ export function useSwimlaneInputResolver(
   const [isLoading, setIsLoading] = useState(false);
 
   const chartWidth$ = useMemo(() => new Subject<number>(), []);
+
+  const selectedJobs$ = useMemo(() => {
+    return getJobsObservable(embeddableInput, anomalyDetectorService, setError).pipe(
+      shareReplay(1)
+    );
+  }, []);
+
+  const bucketInterval$ = useMemo(() => {
+    return combineLatest([selectedJobs$, chartWidth$]).pipe(
+      skipWhile(([jobs, width]) => !Array.isArray(jobs) || !width),
+      map(([jobs, width]) => anomalyTimelineService.getSwimlaneBucketInterval(jobs!, width)),
+      distinctUntilChanged((prev, curr) => {
```

Comment:
If I change the time range on the dashboard, it isn't recalculating the aggregation interval.

![image](https://user-images.githubusercontent.com/7405507/116063551-ffa39c00-a67c-11eb-89e8-afc63c4d914a.png)

Reply:
Good spot, I forgot to track a time range updates from the embeddable input. Fixed in https://github.com/elastic/kibana/pull/98258/commits/f1631b0e1f6e6715d97e3e7d5038a34a609232f3